src/aleph/http/core.clj

```diff
@@ -459,13 +459,16 @@
     (netty/write ch fr)
     (netty/write-and-flush ch empty-last-content)))
 
+(defn- file->stream [^HttpFile file]
+  (-> file
+      (bs/to-byte-buffers {:chunk-size (.-chunk-size file)})
+      s/->source))
+
 (defn send-file-body [ch ssl? ^HttpMessage msg ^HttpFile file]
   (cond
     ssl?
-    (send-streaming-body ch msg
-      (-> file
-        (bs/to-byte-buffers {:chunk-size (.-chunk-size file)})
-        s/->source))
+    (let [body (when (pos-int? (.length file)) (file->stream file))]
+      (send-streaming-body ch msg body))
```

Comment:
I'm trying to think of a way this all might still fail if body becomes nil. According to the [javadoc](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/io/File.html#length()), 0 could also be returned if  it was a device or pipe. I don't think we'd ever support that, though.

Reply:
Wait, if we pass nil into send-streaming-body, won't `(netty/to-byte-buf-stream body' 8192)` throw an exception?

Reply:
The `if-let` is misleading, we are reaching that point:

```clojure
(netty/write-and-flush ch empty-last-content)
```
https://github.com/arnaudgeiser/aleph/blob/559/src/aleph/http/core.clj#L344

Reply:
The aim of this PR is to ensure the client never hangs (waiting for the `TimeoutException`) when the file is empty.
The situation is a little bit better than previously, however, I don't think we are handling every edge case.

Reply:
Ahh, right, right. This code should really be broken up into smaller fns, so to make it easier to follow.